### ISSUE TITLE
Dependent version gets range symbol (^) added with version update.

### ISSUE THREAD

**Issue by rbirkgit** [lerna/lerna#3684]
## Current Behavior

We have two projects in our repo. One is dependent on the other with this dependency:

```
  "dependencies": {
    "@project/api": "15.0.0-beta.7",
    "tslib": "^2.5.0"
  },
```

After a change in @project/api and running `lerna version` the package.json is updated to:

```
  "dependencies": {
    "@project/api": "^15.0.0-beta.8",
    "tslib": "^2.5.0"
  },
```

This is causing big problems for us as when running `yarn install` it will now install the wrong dependency. This because `^` treats any beta release with higher alphabetical name to be considered higher, e.g. `15.0.0-gamma.1`. That's why we specifically removed the `^`, but lerna keeps putting it back!

## Expected Behavior

I expect lerna to honor the `^`, `~` or lack of them and keep the version range the same.

## Steps to Reproduce

Showed above.

### Environment

```
lerna notice cli v6.6.2
lerna info versioning independent

 Environment info:

  System:
    OS: Windows 10 10.0.22000
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
  Binaries:
    Node: 18.14.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 3.2.3 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 9.6.3 - C:\Program Files\nodejs\npm.CMD
  Utilities:
    Git: 2.40.0.
  npmPackages:
    lerna: ^6.6.2 => 6.6.2
```

**Comment by ghiscoding** [comment]
I don't think that is a bug, because internally `^` is used unless you use the `--exact` flag

https://github.com/lerna/lerna/blob/11d847308bb8f4078e47a30a14b4d883948c4530/libs/commands/version/src/index.ts#L125-L126

you can read the docs about [`--exact`](https://github.com/lerna/lerna/tree/main/libs/commands/version#--exact)

> When run with this flag, `lerna version` will specify updated dependencies in updated packages exactly (with no punctuation), instead of as semver compatible (with a `^`).

so in your use case, I guess you do need the `--exact` flag

**Comment by rbirkgit** [comment]
I'm confused. I want it it honor whatever range we use. If we have `~`, `^` or none, it should remain as it was. It works for `~` for example.

**Comment by ghiscoding** [comment]
you're not the first one who brought this up, just doing a quick search and a few similar issues comes up #3383, #3046, #2836 and probably more. #2836 has very similar discussions so no need to add anything more to this conversation
